schema-engine/connectors/sql-schema-connector/src/datamodel_calculator.rs

```diff
@@ -16,8 +16,9 @@ pub fn calculate(schema: &sql::SqlSchema, ctx: &IntrospectionContext, search_pat
     let (schema_string, is_empty, views) = rendering::to_psl_string(&ctx);
     let warnings = warnings::generate(&ctx);
 
-    // Warning codes 5 and 6 are for Prisma 1 default reintrospection.
-    let version = if warnings.iter().any(|w| ![5, 6].contains(&w.code)) {
+    let empty_warnings = warnings.is_empty();
+
+    let version = if !empty_warnings && !warnings.uses_prisma_1_defaults() {
```

Comment:
The original check was super complex to understand. So is this, but hopefully now it reads better. We still need it, remove when removing the version checker.

Reply:
I hear P5 is coming soon